### PR TITLE
[KED-1872] Rename master branch to main and deprecate develop

### PR DESCRIPTION
**Pull request by richardwestenra**
## Description

1. **Rename `master` branch to `main`**
  As '[the Internet Engineering Task Force (IETF) points out](https://tools.ietf.org/id/draft-knodel-terminology-00.html#rfc.section.1.1.1), "Master-slave is an oppressive metaphor that will and should never become fully detached from history" as well as "In addition to being inappropriate and arcane, the master-slave metaphor is both technically and historically inaccurate." There's lots of more accurate options depending on context and it costs me nothing to change my vocabulary, especially if it is one less little speed bump to getting a new person excited about tech.'[[1](https://www.hanselman.com/blog/EasilyRenameYourGitDefaultBranchFromMasterToMain.aspx)]
2. **Deprecate the `develop` branch and just have a single main branch**
  The `develop` branch feels increasingly unnecessary and overcomplicated, both for our release process and our development process in general. So we advocate dropping this branch, and setting `main` as the new default branch for the repository. All PRs will be made against this branch, and when we want to create a new release, we can do it manually via the Github release manager, with a merged/squashed release PR.

## Development notes

I have removed the automatic git tagging from the Makefile version task, as we can handle Github releases via the Github UI instead, rather than relying on automatically generated commit tags. This will give us more flexibility in future, and make it less easy to accidentally screw up a release.

## QA notes

I am also updating the release checklist in Confluence as part of this change.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Review by 921kiyo — APPROVED** [review]
LGTM 👍

File: .circleci/config.yml [hunk 1]
```diff
@@ -210,7 +210,7 @@ commands:
           command: |
             conda activate kedro-viz
             python -m pip install -U pip setuptools wheel
-            pip install git+https://github.com/quantumblacklabs/kedro@master
+            pip install git+https://github.com/quantumblacklabs/kedro@main
```
Comment (921kiyo) [inline .circleci/config.yml:213]: We are changing from `master` to `main`  for Viz only for now, is that correct?

Comment (richardwestenra) [reply .circleci/config.yml]: Yeah that's correct, this was a mistake on my part, thanks for spotting!

File: .circleci/config.yml [hunk 2]
```diff
@@ -307,16 +307,15 @@ workflows:
             - win_build_38
```
Comment (DmitriiDeriabinQB) [inline .circleci/config.yml:307]: To be consistent, in lines 156-160 (out of scope, can't comment there) you could do something like:
```
git init
git checkout -b main
git add -A
git commit -m 'Demo build [ci skip]'
git remote add origin https://github.com/quantumblacklabs/kedro-viz.git
git push origin main:gh-pages --force
```

**Review by DmitriiDeriabinQB — APPROVED** [review]
LGTM

**Review by limdauto — APPROVED** [review]
LGTM!

File: .circleci/config.yml (hunk 2 above)
Comment (richardwestenra) [reply .circleci/config.yml:307]: oh good shout! Thanks :)

**Review by yetudada — APPROVED** [review]
✊🏻✊🏼✊✊🏽✊🏾✊🏿

Looks all good to me!

**Pull request merged**